web/src/components/DatasetsPanel.vue

```diff
@@ -0,0 +1,206 @@
+
+const searchText = ref();
+const filteredDatasets = computed(() => {
+  return props.datasets?.filter((dataset: any) => {
+    return  !searchText.value ||
+    dataset.name.toLowerCase().includes(searchText.value.toLowerCase())
+  });
+});
+const datasetGroups = computed(() => {
+  const groupByKey = "category"
+  const groups: Record<string, Dataset[]> = {};
+  filteredDatasets.value?.forEach((dataset: any) => {
+    const groupName = dataset[groupByKey];
+    if (!groups[groupName]) {
+      groups[groupName] = [];
+    }
+    groups[groupName].push(dataset);
+  });
+  return groups;
+});
+const expandedGroups = ref();
+
+
+function toggleSelected(items: (Dataset | Layer)[]) {
+  // https://stackoverflow.com/a/60334473
+  if (items.every((i) => (<Dataset>i).layers)) {
+    // list of Datasets
+    emit("toggleDatasets", items);
+  } else if (items.every((i) => (<Layer>i).frames)) {
+    // list of Layers
+    items.forEach((item) => {
+      const layer = item as Layer;
+      let name = layer.name;
+      let copy_id = 0;
+      const existing = Object.keys(mapSources.value).filter((sourceId) => {
+        const [layerId] = sourceId.split('.');
+        return parseInt(layerId) === layer.id
+      })
+      if (existing.length) {
+        copy_id = existing.length;
+        name = `${layer.name} (${copy_id})`;
+      }
+      selectedLayers.value = [
+        {...layer, name, copy_id, visible: true, current_frame: 0},
+        ...selectedLayers.value,
+      ];
+    })
+  }
+}
+
+function expandAllDatasets() {
+  if (!expandedGroups.value && filteredDatasets.value) {
+    expandedGroups.value = Object.keys(datasetGroups.value)
+  }
+}
+
+onMounted(expandAllDatasets);
+
+watch(filteredDatasets, expandAllDatasets)
+</script>
+
+<template>
+  <div class="panel-content-outer with-search">
+    <v-text-field
+      v-model="searchText"
+      label="Search Datasets"
+      variant="outlined"
+      density="compact"
+      class="mb-2"
+      append-inner-icon="mdi-magnify"
+      hide-details
+    />
+    <v-card class="panel-content-inner">
+      <v-expansion-panels
+        v-if="props.datasets?.length"
+        v-model="expandedGroups"
+        variant="accordion"
+        class="dataset-list"
+        multiple
+      >
+        <v-expansion-panel
+          v-for="[groupName, groupDatasets] in Object.entries(datasetGroups)"
+          :key="groupName"
+          :value="groupName"
+          bg-color="background"
+        >
+          <v-expansion-panel-title style="font-weight: bold" class="capitalize secondary-text">
+            {{ groupName }}
+          </v-expansion-panel-title>
+          <v-expansion-panel-text>
+            <v-treeview
+              :items="groupDatasets"
+              item-value="id"
+              item-title="name"
+              item-children="layers"
+              open-on-click
+            >
+              <template v-slot:title="{title, item}">
+                <div class="dataset-title">
```

Comment:
Having the whole row clickable seems more intuitive to me. Since there's no other action that can be taken on that row, forcing the user to click a small icon seems unnecessary.

```suggestion
                <div class="dataset-title" @click="toggleSelected([item])">
```

Reply:
Also it seems the class `dataset-title` is misleading, since this is actually displaying a layer, not a dataset.

Reply:
Ok, I changed the structure (and class name) of the item title here to make the whole row clickable for adding a layer to selected layers: [52dc78d](https://github.com/OpenGeoscience/uvdat/pull/102/commits/52dc78da5aaf709d058d5442a02b0a707682db32). It required a bit of restructuring because we don't want the whole row to be clickable in the case where a Dataset has a checkbox next to it in the Project config context. In that case, clicking on the row expands the Dataset to show its Layers, while clicking on the checkbox toggles the selected state.

Reply:
This is some more anecdotal evidence towards what I'm proposing in  https://github.com/OpenGeoscience/uvdat/pull/102#discussion_r1978298094, FWIW.